File: 2023/src/util/input.py
def get_input_array() -> list[str]:
    with open("input.txt") as file:
        return [line.strip() for line in file]


def get_input_matrix() -> list[list[str]]:
    return [list(line) for line in get_input_array()]


def get_split_input(delimiter: str) -> list[str]:
    with open("input.txt") as file:
        return file.read().split(delimiter)


def get_raw_input() -> str:
    with open("input.txt") as file:
        return file.read()

File: 2024/src/utils/input.ts
import { readFileSync } from 'node:fs';

export function getRawInput(): string {
  return readFileSync('input.txt', 'utf8');
}

export function getInput() {
  return getRawInput().trim();
}

export function getSplitInput(delimiter = '\n') {
  return getInput().split(delimiter);
}

export function getInputLines() {
  return getSplitInput().map(l => l.trim());
}

export function getInputMatrix(delimiter = '') {
  return getInputLines().map(l => l.split(delimiter));
}

export function getNumberMatrix(delimiter = '') {
  return getInputMatrix(delimiter).map(line => line.map(Number));
}
